src/psyclone/tests/psyGen_test.py

```diff
@@ -183,6 +183,83 @@ def apply(self, _1):
     assert trans.name == "TestTrans"
 
 
+def test_transformation_get_options():
+    ''' Test that the get_option method behaves in the
+    expected way.'''
+    class TestTrans(Transformation):
+        '''Utilty transformation to test methods of the abstract
+        Transformation class.'''
+        def apply(self, node, valid: bool = True):
+            pass  # pragma: no cover
+    trans = TestTrans()
+    assert trans.get_option("valid", valid=True)
+
+    with pytest.raises(ValueError) as excinfo:
+        trans.get_option("invalid")
+    assert ("option 'invalid' is not a valid option for 'TestTrans'. "
+            "Valid options are '['valid']." in str(excinfo.value))
+
+
+def test_transformation_apply_deprecation_message(capsys):
+    '''Test that passing the options dict to the Transformation.apply
+    function gets the expected deprecation message.'''
+    class TestTrans(Transformation):
+        '''Utility transformation to test methods.'''
+        def apply(self, node=None, options=None):
+            super().apply(node, options=options)
+
+    instance = TestTrans()
+    instance.apply(options={"dict": True})
+    out, err = capsys.readouterr()
+    assert ("Deprecation Warning: The options parameter to "
+            "Transformation.apply is now deprecated. Please use "
+            "the individual arguments, or unpack the options with "
+            "**options. See the documentation for more details." in out)
+
+
+def test_transformation_get_valid_options():
+    '''Test that the get_valid_options method behaves in the expected
+    way.'''
+    class TestTrans(Transformation):
+        '''Utilty transformation to test methods of the abstract
+        Transformation class.'''
+        def apply(self, node, valid: bool = True, untyped=False):
+            pass  # pragma: no cover
+
+    options = TestTrans.get_valid_options()
+    assert options['valid'].default
+    assert options['valid'].type is bool
+    assert options['valid'].typename == "bool"
+    assert options['untyped'].default is False
+    assert options['untyped'].type is None
+    assert options['untyped'].typename is None
+
```

Comment:
Please add a test showing that valid options are inherited correctly from parent transformation classes.

Reply:
Added